File: Lab1/cs_solution/Assistant.cs
public class Assistant
{
    public string Name; 
    private List<Display> assignedDisplays;

    public Assistant(string name)
    {
        Name = name;
        assignedDisplays = new List<Display>();
    }

    public void AssignDisplay(Display display)
    {
        assignedDisplays.Add(display);
        Console.WriteLine($"{display.Model} assigned to {Name}.\n");
    }

    public void Assist()
    {
        Console.WriteLine("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        Console.WriteLine($"Assisting with display comparisons for {Name}:");
        Console.WriteLine("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        for (int i = 0; i < assignedDisplays.Count - 1; i++)
        {
            assignedDisplays[i].CompareWithMonitor(assignedDisplays[i + 1]);
            Console.WriteLine();
        }
    }

    public Display? BuyDisplay(Display display)
    {
        if (assignedDisplays.Remove(display))
        {
            Console.WriteLine($"{display.Model} has been removed from the list");
            return display;
        }
        else
        {
            Console.WriteLine($"{display.Model} is not in the list.\n");
            return null;
        }
    }
}

File: Lab1/ts_solution/src/modules/Assistant.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Assistant = void 0;
var Assistant = /** @class */ (function () {
    function Assistant(asName) {
        this.Name = asName;
        this.assignedDisplays = [];
    }
    Assistant.prototype.AssignDisplay = function (display) {
        this.assignedDisplays.push(display);
        console.log("".concat(display.Model, " assigned"));
    };
    Assistant.prototype.Assist = function () {
        console.log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        console.log("Assisting with display comparisons for ".concat(this.Name, ":"));
        console.log("~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~");
        for (var i = 0; i < this.assignedDisplays.length - 1; i++) {
            this.assignedDisplays[i].CompareWithMonitor(this.assignedDisplays[i + 1]);
            console.log("\n");
        }
    };
    Assistant.prototype.BuyDisplay = function (display) {
        var index = this.assignedDisplays.indexOf(display);
        if (index !== -1) {
            this.assignedDisplays.splice(index, 1);
            console.log("".concat(display.Model, " has been removed from the list"));
            return display;
        }
        else {
            console.log("".concat(display.Model, " is not in the list.\n"));
            return null;
        }
    };
    return Assistant;
}());
exports.Assistant = Assistant;
